pkg/crc/image/image.go

```diff
@@ -0,0 +1,112 @@
+package image
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"runtime"
+	"strings"
+
+	crcpreset "github.com/code-ready/crc/pkg/crc/preset"
+	"github.com/containers/image/v5/copy"
+	"github.com/containers/image/v5/signature"
+	"github.com/containers/image/v5/transports/alltransports"
+	"github.com/containers/image/v5/types"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+var (
+	registryURI = "quay.io"
+	orgName     = "crcont"
+)
+
+type OCIImageHandler struct {
+	ImageName string
+	ImageTag  string
+	Progress  io.Writer
+}
+
+func (img *OCIImageHandler) uri() string {
+	return strings.Join([]string{registryURI, orgName, img.ImageName}, "/")
+}
+
+func (img *OCIImageHandler) parse(image string) (types.ImageReference, error) {
+	return alltransports.ParseImageName(image)
+}
+
+func (img *OCIImageHandler) policyContext() (*signature.PolicyContext, error) {
+	policy := &signature.Policy{Default: []signature.PolicyRequirement{signature.NewPRInsecureAcceptAnything()}}
+	policyContext, err := signature.NewPolicyContext(policy)
+	if err != nil {
+		return nil, fmt.Errorf("error creating security context: %s", err.Error())
+	}
+
+	return policyContext, nil
+}
+
+func (img *OCIImageHandler) getSystemContext() *types.SystemContext {
+	return &types.SystemContext{
+		OSChoice: "linux",
+	}
+}
+
+// CopyImage pull the image from the registry and put it to local storage
+func (img *OCIImageHandler) CopyImage(destPath string) ([]byte, error) {
+	// Source Image from docker transport
+	srcImg := fmt.Sprintf("docker://%s:%s", img.uri(), img.ImageTag)
+	srcRef, err := img.parse(srcImg)
+	if err != nil {
+		return nil, fmt.Errorf("invalid source image name %s: %v", srcImg, err)
+	}
+
+	destImg := fmt.Sprintf("dir:%s", destPath)
+	destRef, err := img.parse(destImg)
+	if err != nil {
+		return nil, fmt.Errorf("invalid destination name %s: %v", destImg, err)
+	}
+
+	policyContext, err := img.policyContext()
+	if err != nil {
+		return nil, err
+	}
+
+	return copy.Image(context.Background(), policyContext,
+		destRef, srcRef, &copy.Options{
+			ReportWriter: img.Progress,
+			SourceCtx:    img.getSystemContext(),
+		})
+}
+
+func GetBundleImageLayerFromManifest(manifestData []byte) (string, error) {
+	manifest := &v1.Manifest{}
+	// Get the file layer sha from manifest
+	if err := json.Unmarshal(manifestData, manifest); err != nil {
+		return "", err
+	}
+	return removeShaFromDigestString(manifest.Layers[0].Digest.String()), nil
+}
+
+func removeShaFromDigestString(digestString string) string {
+	return strings.TrimPrefix(digestString, "sha256:")
+}
+
+func GetImageName(preset crcpreset.Preset) string {
+	img := imageNameForOs(preset)
+	return img[runtime.GOOS]
+}
+
+func imageNameForOs(preset crcpreset.Preset) map[string]string {
+	if preset == crcpreset.Podman {
```

Comment:
Dunno if this method should be here or in `pkg/crc/constants/constants.go` with other similar code?

Reply:
I prefer it here but don't have any strong opinion to move (if you really think)